### PR TITLE
tailcfg: add UserProfile.Groups

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#13375

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -175,6 +175,21 @@ type UserProfile struct {
 	// Roles exists for legacy reasons, to keep old macOS clients
 	// happy. It JSON marshals as [].
 	Roles emptyStructJSONSlice
+
+	// Groups contains group identifiers for any group that this user is
+	// a part of and that the coordination server is configured to tell
+	// your nodes about. (Thus, it may be empty or incomplete.)
+	// There's no difference between a nil vs an empty list.
+	// The list is always be sorted.
```
Comment (maisem) [inline tailcfg/tailcfg.go:183]: Grammar

**Review by maisem — APPROVED** [review]

File: tailcfg/tailcfg.go (hunk 1 above)
Comment (bradfitz) [reply tailcfg/tailcfg.go]: words

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -175,6 +175,21 @@ type UserProfile struct {
 	// Roles exists for legacy reasons, to keep old macOS clients
 	// happy. It JSON marshals as [].
 	Roles emptyStructJSONSlice
+
+	// Groups contains group identifiers for any group that this user is
```
Comment (maisem) [inline tailcfg/tailcfg.go:179]: Could also bump the cap ver so that we don't send it down to clients who don't understand it

Comment (bradfitz) [reply tailcfg/tailcfg.go:180]: done

**Pull request merged**